core/controllers/base.py

```diff
@@ -295,8 +299,81 @@ def dispatch(self):
                 self.handle_exception(e, self.app.debug)
                 return
 
+        schema_validation_succeeded = True
+        try:
+            self.validate_args_schema()
+        except self.InvalidInputException as e:
+            self.handle_exception(e, self.app.debug)
+            schema_validation_succeeded = False
+        # TODO(#13155): Remove this clause once all the handlers have had
+        # schema validation implemented.
+        except NotImplementedError as e:
+            self.handle_exception(e, self.app.debug)
+            schema_validation_succeeded = False
+
+        if schema_validation_succeeded is False:
+            return
+
         super(BaseHandler, self).dispatch()
 
+    def validate_args_schema(self):
+        """Validates schema for controller layer handler class arguments.
+
+        Raises:
+            InvalidInputException. Schema validation failed.
+            NotImplementedError. Schema is not provided in handler class.
+        """
+        handler_class_name = self.__class__.__name__
+        request_method = self.request.environ['REQUEST_METHOD']
+        handler_args = self.payload
```

Comment:
Got your point, I have defined handler_args for payloads / requests around the same line.
I have shifted handler_args assignment in the top, since for debugging purposes one may want to print all the data coming to a handler, so every variable used by validate_args_schema() should be on the top, otherwise they will face error.
Link for debugging section for schema validation in wiki page: https://github.com/oppia/oppia/wiki/Validation-of-handler-args#debugging-tricks
**Question**: Please explain the point for handling GET and DELETE separately as because both of the request methods receive arguments from ```self.request```.